packages/amplify_api/test/amplify_api_graphql_helpers_test.dart

```diff
@@ -15,15 +15,23 @@
 
 import 'package:flutter/foundation.dart';
 import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api/src/graphql/graphql_response_decoder.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 import 'resources/Blog.dart';
 import 'resources/ModelProvider.dart';
 
 void main() {
-  test("ModelQueries.get() should craft a valid request", () {
-    AmplifyAPI api = AmplifyAPI(modelProvider: ModelProvider.instance);
+  AmplifyAPI? api;
+  setUp(() {
+    api = AmplifyAPI(modelProvider: ModelProvider.instance);
+  });
 
+  tearDown(() {
+    api = null;
+  });
```

Comment:
This block is not used by the tests (can be removed). If you do need an instance for these tests, I would just declare it once, since setup/teardown should be used more for like initialization/de-initialization logic which needs to be run before/after tests like if you're resetting properties on an object or if initializing an object is asynchronous. Here, the instance won't change between tests, though. And as you do later, you create a separate local instance for that specific test.

Another way you could do this is with groups, since you have a clear distinction of using `AmplifyAPI` with a model provider and without.

```dart
void main() {
  group('with ModelProvider', () {
    final api = AmplifyAPI(modelProvider: ModelProvider.instance);

    test('...', () {});
    test('...', () {});
    test('...', () {});
  });

  group('without ModelProvider', () {
    final api = AmplifyAPI();

    test('...', () {});
    test('...', () {});
    test('...', () {});
  });
}
```

Reply:
Awesome, I love this.